steward/src/commands/cosmos_mode.rs

```diff
@@ -33,15 +33,19 @@ impl Runnable for CosmosSignerCmd {
                 });
 
             // Configure TLS
-            let tls_config = server::load_server_config(config)
+            let tls_config = server::load_server_config(&config)
                 .await
                 .unwrap_or_else(|err| {
                     status_err!("failed to load TLS config: {}", err);
                     std::process::exit(1)
                 });
 
             // run it
-            let addr = ([127, 0, 0, 1], STEWARD_PORT).into();
+            let port = match config.server.port {
+                Some(port) => port,
+                None => DEFAULT_STEWARD_PORT,
+            };
+            let addr = ([127, 0, 0, 1], port).into();
```

Comment:
To answer a question I believe you asked out of band, yes, we should also make the listen address configurable.

Reply:
Including domain name or just IP?